c++/src/kj/test-helpers.c++

```diff
@@ -32,13 +32,8 @@ namespace kj {
 namespace _ {  // private
 
 bool hasSubstring(StringPtr haystack, StringPtr needle) {
-  // TODO(perf): This is not the best algorithm for substring matching.
   if (needle.size() <= haystack.size()) {
-    for (size_t i = 0; i <= haystack.size() - needle.size(); i++) {
-      if (haystack.slice(i).startsWith(needle)) {
-        return true;
-      }
-    }
+    return strstr(haystack.cStr(), needle.cStr()) != nullptr;
```

Comment:
Technically this breaks support for strings with NUL bytes in them... did you verify it's never used with NUL bytes?

Reply:
No I didn't try NUL bytes as I clearly didn't think of it (although apparently none of the in-tree tests require this). I'll switch to memmem on Linux then.

Reply:
Admittedly NUL bytes are probably never used in test strings, but OTOH the performance of this helper probably doesn't really matter since it's just for tests.